rasterio/rio/warp.py

```diff
@@ -309,20 +310,14 @@ def warp(
 
             else:
                 dst_crs = src.crs
-                inv_transform = ~src.transform
                 eps = sys.float_info.epsilon
-                c1, r1 = inv_transform * (left + eps, top + eps)
-                c2, r2 = inv_transform * (right + eps, top + eps)
-                c3, r3 = inv_transform * (right + eps, bottom + eps)
-                c4, r4 = inv_transform * (left + eps, bottom + eps)
-                col1 = min(c1, c2, c3, c4)
-                col2 = max(c1, c2, c3, c4)
-                row1 = min(r1, r2, r3, r4)
-                row2 = max(r1, r2, r3, r4)
-                col1 = floor(col1)
-                col2 = ceil(col2)
-                row1 = floor(row1)
-                row2 = ceil(row2)
+                rows = np.array([top, top, bottom, bottom]) + eps
+                cols = np.array([left, right, right, left]) + eps
+                rows, cols = rowcol(src.transform, rows, cols)
```

Comment:
The default for rowcol is to floor results. The tests pass. 
Oddly, when trying to exactly match the old behavior (op=float and doing the floor/ceil calls), the `test_warp_no_reproject` test fails.